osu.Game/Screens/Select/Footer.cs

```diff
@@ -41,21 +39,16 @@ public class Footer : Container
         /// <para>Higher depth to be put on the left, and lower to be put on the right.</para>
         /// <para>Notice this is different to <see cref="Options.BeatmapOptionsOverlay"/>!</para>
         /// </param>
-        public void AddButton(string text, Color4 colour, Action action, Key? hotkey = null, float depth = 0)
+        public void AddButton(FooterButton button, string text, Color4 colour, Action action, Key? hotkey = null, float depth = 0)
```

Comment:
If the caller has to create the ``FooterButton`` themselves then they should probably also set the other parameters of this method in the constructor...

Reply:
I just left `.AddButton()` mostly as is since the buttons share the same properties.

Reply:
Have applied this review myself.